### PR TITLE
refactor: backend seq_lens ownership and fix dsv3

### PR DESCRIPTION
**Pull request by FC-Li**
- Drop the shared seq_lens_buf parameter from every backend's init_cuda_graph_state; each backend now owns its cache-seqlens buffer and copies the live lengths in at replay, removing the dependency on the controller mutating a shared tensor. Updates base/mha/mla/trtllm/trtllm_mla/ tokenspeed_mla/dsa/inkling/hybrid_linear_attn/deepseek_v4 and the cuda-graph wrapper + affected tests.
- Fall back to plain all-gather + torch.argmax when CuTe DSL distributed argmax is unavailable (e.g. H20 lacks TMA cluster launch), fixing the spec-decode logits_processor assertion crash.
- Pass device_id to init_process_group on CUDA so PyTorch binds the process group eagerly and stops emitting the c10d barrier() device warning.
- Keep the FlashMLA new-API compatibility (drop num_splits, lazy tile sched meta) and the fp8 contiguous kv_a fix;

### CODE REVIEW

File: python/tokenspeed/runtime/layers/attention/backends/flashmla.py [hunk 1]
```diff
@@ -381,86 +375,32 @@ def init_forward_metadata_capture_cuda_graph(
...
-            )
-        else:
+        if not (decode_no_spec or is_target_verify or is_draft_extend):
             raise RuntimeError(f"Not supported forward mode: {forward_mode}")
 
+        self.forward_decode_metadata = FlashMLADecodeMetadata(
+            num_extends=0,
+            flashmla_metadata=self._capture_decode_tile_metadata(bs),
+            block_table=self.cuda_graph_kv_indices[:bs],
+            seq_lens_k=self.cuda_graph_seq_lens_k[:bs],
```
[the remark this answers is not in the document]
Comment (FC-Li) [reply python/tokenspeed/runtime/layers/attention/backends/flashmla.py:390]: @nperrin-fr This claim is wrong.  We maybe don't really need the code following 
"""
        # Seed before building the tile schedule: it is recorded against these
        # lengths, and the capture run reads them before any replay.
        self.cuda_graph_seq_lens_k[:bs].copy_(seq_lens[:bs])
 """

[the remark this answers is not in the document]
Comment (nperrin-fr) [reply python/tokenspeed/runtime/layers/attention/backends/flashmla.py:390]: yeah, actually `seq_lens: torch.Tensor` could also be taken off, let me check
